server/routes/user.js

```diff
@@ -0,0 +1,109 @@
+const express = require("express");
+const router = express.Router();
+
+const auth = require("../middleware/auth");
+
+const bcrypt = require("bcryptjs");
+const jwt = require("jsonwebtoken");
+
+const {
+  validate,
+  registerValidationRules,
+  loginValidationRules
+} = require("../middleware/validator");
+const keys = require("../config/keys");
+
+const User = require("../models/User");
+
+// @route  POST user/register
+// @desc   register user
+// @access public
+
+router.post(
+  "/register",
+  registerValidationRules(),
+  validate,
+  async (req, res) => {
```

Comment:
quick question about the middleware, does the response of each middleware function gets passed into the next() right? Sorry am relatively newish to express, Also, nice async usage, I've been trying to incorporate that in my routes for code consistency,

Reply:
1. registerValidationRules() -> this function returns the array (validation on each field)
2. validate -> this is the middleware function that validates the fields based on the rules, explicitly need to specify next() to continue